### PR TITLE
Implement the set-pool-size action

### PR DESCRIPTION
**Pull request by lmlg**

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -161,6 +162,22 @@ def _add_osd_action(self, event: ActionEvent):
...
 
+    def _set_pool_size_action(self, event: ActionEvent):
+        """Set the size for one or more pools."""
+        pools = event.params.get("pools")
+        size = event.params.get("size")
+
+        cmd = ["sudo", "microceph", "pool", "set-rf", "--size", str(size), pools]
+        try:
+            logging.debug("Setting the new pool(s) size")
+            subprocess.run(cmd, check=True, timeout=180)
```
Comment (UtkarshBhatthere) [inline src/charm.py:173]: The `_run_cmd` method in `microceph.py` has inbuilt logging for success and failure cases and ideally I'd prefer all CLI command calls to be organised in that file as that makes it simpler for mocking and keeps logically bound methods together.

**Review by UtkarshBhatthere — CHANGES_REQUESTED** [review]
The change LGTM, one minor nit regarding code organisation.

File: src/charm.py (hunk 1 above)
Comment (lmlg) [reply src/charm.py]: Good idea. I've refactored things a bit.

File: src/charm.py [hunk 2]
```diff
@@ -110,6 +111,18 @@ def _on_stop(self, event: ops.StopEvent):
...
 
+    def _set_pool_size_action(self, event: ops.framework.EventBase) -> None:
+        """Set the size for one or more pools."""
+        pools = event.params.get("pools")
+        size = event.params.get("size")
+
+        try:
+            microceph.set_pool_size(pools, size)
+            event.set_results({"status": "success"})
+        except Exception:
```
Comment (UtkarshBhatthere) [inline src/charm.py:122]: This is mostly a nit but i feel catching base `Exception` class may also prevent exposing other issues (like Timeout exceptions and what not)

**Review by UtkarshBhatthere — CHANGES_REQUESTED** [review]
single nit

**Review by UtkarshBhatthere — DISMISSED** [review]
LGTM, let's ship it!

**Review by UtkarshBhatthere — APPROVED** [review]
LGTM, let's ship it this time!

**Pull request merged**